### DISCUSSION TITLE
Audio pronunciation

### DISCUSSION THREAD

**Discussion by getvoice** [axelboc/anki-ultimate-geography#604] [Ideas]
Hi!

I've been using the deck for the last several weeks, and it's great. One thing I found myself doing a lot, especially when I got to African countries, is checking the pronunciation.  I was going to add the pronunciation audio field to the deck, and then I thought I might also offer it for the creator(s) of the deck to implement in the next version. I can do all the work of adding the mp3 files and everything since I was going to do it for my deck anyway. I don't know how to contact the creator directly here and for that matter who is the creator? I assumed it's @axelboc (??)

**Comment by axelboc** [comment]
Hi @getvoice, thanks for offering to help out! There's actually already an issue dedicated to adding audio pronunciation to the deck: https://github.com/anki-geo/ultimate-geography/issues/81 -- as you will see, it has been discussed at length and various issues have been uncovered (audio vs phonetic, automatic vs manual, which accent, etc.) which have hindered implementation.

A way forward might appear as we expand the Ultimate Geography deck into a broader geography deck ecosystem that would allow users to build their own decks from various extension repositories. See #603 for the latest discussions on this topic. What this means is that you'd be able to create an extension repository dedicated to audio pronunciation (in one or more accents and/or languages) and users would be able to build a deck that includes audio pronunciation by simply referencing your repo and selecting one of the languages/accents available. 

We're still a long way off from this, though, so not sure what recommendation to give you at this point. If you add a new field to your UG deck, it will make it [difficult to upgrade](https://github.com/anki-geo/ultimate-geography#changes-that-prevent-upgrading-). You could update the templates to include auto-generated audio from an online service on every note, but this won't give you the best results... You could fork this repo, add an audio pronunciation field, build the deck yourself, and then keep your fork up to date with the main repo, but you may have to resolve tricky git conflicts down the line.

There's no perfect option at the moment, but forking the repo means that the audio pronunciation files you select will at least be version-controlled and therefore easier to migrate to a separate extension repo when the time comes.

**Reply by getvoice** [reply]
Thank you for the reply!
